### ISSUE TITLE
Flat storage total size of cached deltas exceeded expected limits

### ISSUE THREAD

**Issue by YuXiaoCoder** [near/nearcore#10757]
**Describe the bug**
What should I do to get rid of these warning logs?
```log
2024-03-12T02:34:53.547094Z  INFO stats: Catchups
Sync block 3eitm2GGAGhmvg6k7bFnsabSjkDC1WDPxH3buxZofTiN@114537108
Shard sync status: Shard 0 done, Shard 1 done, Shard 2 done, Shard 3 done
Next blocks to catch up:
2024-03-12T02:34:54.852350Z  WARN chain: Flat storage total size of cached deltas exceeded expected limits shard_id=2 flat_head_height=114537107 cached_deltas=12273 cached_changes_size_bytes=910.1 MB
2024-03-12T02:34:54.860466Z  WARN chain: Flat storage total size of cached deltas exceeded expected limits shard_id=4 flat_head_height=114537107 cached_deltas=12273 cached_changes_size_bytes=167.2 MB
2024-03-12T02:34:57.224028Z  WARN chain: Flat storage total size of cached deltas exceeded expected limits shard_id=2 flat_head_height=114537107 cached_deltas=12274 cached_changes_size_bytes=910.1 MB
2024-03-12T02:34:57.232258Z  WARN chain: Flat storage total size of cached deltas exceeded expected limits shard_id=4 flat_head_height=114537107 cached_deltas=12274 cached_changes_size_bytes=167.4 MB
2024-03-12T02:34:58.632276Z  WARN chain: Flat storage total size of cached deltas exceeded expected limits shard_id=2 flat_head_height=114537107 cached_deltas=12275 cached_changes_size_bytes=910.1 MB
2024-03-12T02:34:58.638957Z  WARN chain: Flat storage total size of cached deltas exceeded expected limits shard_id=4 flat_head_height=114537107 cached_deltas=12275 cached_changes_size_bytes=167.4 MB
2024-03-12T02:34:59.819277Z  WARN chain: Flat storage total size of cached deltas exceeded expected limits shard_id=2 flat_head_height=114537107 cached_deltas=12276 cached_changes_size_bytes=910.1 MB
2024-03-12T02:34:59.825716Z  WARN chain: Flat storage total size of cached deltas exceeded expected limits shard_id=4 flat_head_height=114537107 cached_deltas=12276 cached_changes_size_bytes=167.4 MB
2024-03-12T02:35:00.607253Z  INFO network: Closing connection to Some(ed25519:8EmwKNs4JaWCLuR7CSs7vhwRcZV1nRJ36vE2jmABUX6c@169.155.47.142:24567) err=Recv(IO(Kind(UnexpectedEof)))
2024-03-12T02:35:00.608050Z  INFO network: ed25519:pstWk6sk3uA11V19jV1EP2WX8kdi2PXxa5dwf4aP19K: Peer Some(ed25519:8EmwKNs4JaWCLuR7CSs7vhwRcZV1nRJ36vE2jmABUX6c@169.155.47.142:24567) disconnected, reason: stream error
2024-03-12T02:35:01.487381Z  WARN chain: Flat storage total size of cached deltas exceeded expected limits shard_id=2 flat_head_height=114537107 cached_deltas=12277 cached_changes_size_bytes=910.1 MB
2024-03-12T02:35:01.498586Z  WARN chain: Flat storage total size of cached deltas exceeded expected limits shard_id=4 flat_head_height=114537107 cached_deltas=12277 cached_changes_size_bytes=167.5 MB
2024-03-12T02:35:01.533560Z ERROR network: Failed to save peer data err=Peer ed25519:8EmwKNs4JaWCLuR7CSs7vhwRcZV1nRJ36vE2jmABUX6c is missing in the peer store
```

**To Reproduce**
/opt/core/refiner -c /conf/refiner/refiner.json run

refiner.json
```json
{
    "refiner": {
        "chain_id": 1313161554,
        "engine_path": "/mnt/auroramain/node/engine",
        "tx_tracker_path": "/mnt/auroramain/node/engine/tx_tracker",
        "engine_account_id": "aurora"
    },
    "input_mode": {
        "Nearcore": {
            "path": "/mnt/auroramain/node/near"
        }
    },
    "output_storage": {
        "path": "/mnt/auroramain/node/refiner",
        "batch_size": 10000
    },
    "socket_server": {
        "path": "/mnt/auroramain/node/refiner.sock"
    }
}
```

**Expected behavior**
No more warning logs

**Screenshots**
![image](https://github.com/near/nearcore/assets/26297513/59444b04-d585-475a-9764-e668f50c9823)

**Version (please complete the following information):**
- nearcore https://github.com/aurora-is-near/borealis-engine-lib/releases/tag/0.26.2
- mainnet

**Comment by robin-near** [comment]
Thanks for reporting! Should be fixed by #10756 included in 1.37.2.

**Comment by YuXiaoCoder** [comment]
Successfully resolved
